packages/data-mate/src/adapters/date-frame-adapter/index.ts

```diff
@@ -0,0 +1,210 @@
+import { FieldType, DataTypeFieldConfig } from '@terascope/types';
+import { isNil } from 'lodash';
+import { Column, ColumnTransformConfig } from '../../column';
+import { DataFrame } from '../../data-frame';
+import {
+    FieldTransformConfig, isFieldTransform, isFieldValidation, ProcessMode,
+    FieldValidateConfig, FunctionDefinitions, isFieldOperation
+} from '../../interfaces';
+
+import {
+    ColumnValidateConfig, TransformType, TransformMode,
+} from '../../column/interfaces';
+
+import {
+    VectorType,
+} from '../../vector/interfaces';
+
+// @TODO: fix this
+
+const FieldTypeToVectorDict: Record<FieldType, VectorType> = {
+    [FieldType.String]: VectorType.String,
+    [FieldType.Text]: VectorType.String,
+    [FieldType.Keyword]: VectorType.String,
+    [FieldType.KeywordCaseInsensitive]: VectorType.String,
+    [FieldType.KeywordTokens]: VectorType.String,
+    [FieldType.KeywordTokensCaseInsensitive]: VectorType.String,
+    [FieldType.KeywordPathAnalyzer]: VectorType.String,
+    [FieldType.Domain]: VectorType.String,
+    [FieldType.Hostname]: VectorType.String,
+    [FieldType.NgramTokens]: VectorType.String,
+    [FieldType.IP]: VectorType.IP,
+    [FieldType.IPRange]: VectorType.IPRange,
+    [FieldType.Date]: VectorType.Date,
+    [FieldType.Boolean]: VectorType.Boolean,
+    [FieldType.Float]: VectorType.Float,
+    [FieldType.Number]: VectorType.Int,
+    [FieldType.Byte]: VectorType.Int,
+    [FieldType.Short]: VectorType.Int,
+    [FieldType.Integer]: VectorType.Int,
+    [FieldType.Long]: VectorType.Boolean,
+    [FieldType.Double]: VectorType.Boolean,
```

Comment:
We can't technically support double until BigFloat but float is fine for now

```suggestion
    [FieldType.Double]: VectorType.Float,
```